### PR TITLE
Fix visiblity of private threads and communities on user profiles

### PR DESCRIPTION
**Pull request by brianlovin**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- api

Will need to do a prod cut with this asap @mxstbr

### CODE REVIEW

File: api/models/community.js [hunk 1]
```diff
@@ -74,6 +75,76 @@ export const getCommunitiesByUser = (userId: string): Promise<Array<DBCommunity>
...
+    // get all the user's communities
+    .getAll(evaluatingUserId, { index: 'userId' })
+    // only return communities the user is a member of
+    .filter({ isMember: true })
+    // get the community objects for each community
+    .eqJoin('communityId', db.table('communities'))
+    // get rid of unnecessary info from the usersCommunities object on the left
+    .without({ left: ['id', 'communityId', 'userId', 'createdAt'] })
+    // zip the tables
+    .zip()
```
Comment (ryota-murakami) [inline api/models/community.js:91]: `zip() ` 🤔
is ti RethinkDB term😅❓

Comment (mxstbr) [reply api/models/community.js:91]: Yes, this is the RethinkDB querying API :blush:

**Review by mxstbr — DISMISSED** [review]
Sure, LGTM

File: api/models/community.js (hunk 1 above)
Comment (ryota-murakami) [reply api/models/community.js:91]: i'm apreciete🙏
i'll more learn a lot

Comment (brianlovin) [reply api/models/community.js:91]: @ryota-murakami https://rethinkdb.com/api/javascript/eq_join/ is helpful here :)

Comment (ryota-murakami) [reply api/models/community.js:91]: @brianlovin wow, thank you so much😄and early morning!🍞☕️🌄

**Comment by brianlovin** [comment]
CI failure is the memory leak problem :(

**Comment by ryota-murakami** [comment]
about CI failure, my solution looks like fine😅 
@mxstbr could you please review? when you are not busy🙇‍♂️ https://github.com/withspectrum/spectrum/pull/3121

**Pull request merged**